app/models/book.rb

```diff
@@ -5,11 +5,22 @@ class Book < ApplicationRecord
   validates :title, presence: true, uniqueness: { case_sensitive: false}
   validates :author, presence: true
 
-  def self.available
-    includes(:checkouts).where(checkouts: { id: nil })
+  def self.unavailable
+    Book.joins(:checkouts).where(checkouts: {closed_at: nil })
   end
 
-  def self.unavailable
-    includes(:checkouts).where.not(checkouts: { id: nil })
+  def self.available
+    # I needed to use a LEFT JOIN because
+    # `Book.joins(:checkouts).where.not(checkouts: {closed_at: nil })`
+    # displayed checkouts where ANY checkout associated with the book
+    # and user would be listed as unavailable. I need books to listed as
+    # available when EVERY checkout is closed. 
+    Book.joins(
```

Comment:
@paulcsmith, I used a `LEFT JOIN` because `Book.joins(:checkouts).where.not(checkouts: {closed_at: nil })` displayed books as available if ANY of the checkouts associated with that book_id and user_id where marked as closed. I need books to be listed as available only if EVERY checkout is marked as closed. 

Details: The inner query finds all checkouts yet to be returned.  Then I join them with all books. Using a `LEFT JOIN` means that if a book is not checked out, it will still be included in the results table, just with nil values for the checkout.

Any advice on how to make the SQL match a Rails construct would be great.